pkg/k8shandler/deployment.go

```diff
@@ -472,6 +456,51 @@ func (node *deploymentNode) update(upgradeStatus *api.ElasticsearchNodeStatus) e
 	return nil
 }
 
+func (node *deploymentNode) executeUpdate() error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// isChanged() will get the latest revision from the apiserver
+		// and return false if there is nothing to change and will update the node object if required
+		if node.isChanged() {
+			if updateErr := sdk.Update(&node.self); updateErr != nil {
+				logrus.Debugf("Failed to update node resource %v: %v", node.self.Name, updateErr)
+				return updateErr
+			}
+		}
+		return nil
+	})
+}
+
+func (node *deploymentNode) updateStuckPod(upgradeStatus *api.ElasticsearchNodeStatus) (bool, error) {
+	pods, err := GetPodsForDeployment(node.self.Name, node.self.Namespace)
```

Comment:
why are we getting pods instead of just iterating our `Status` object for the cluster?